app/code/Magento/Catalog/view/frontend/templates/product/list/items.phtml

```diff
@@ -223,14 +223,21 @@ switch ($type = $block->getType()) {
                                             <?php if ($showCart):?>
                                                 <div class="actions-primary">
                                                     <?php if ($_item->isSaleable()):?>
+                                                        <?php $postDataHelper =
+                                                            $this->helper
+                                                            (Magento\Framework\Data\Helper\PostHelper::class);
+                                                        $postData =
+                                                            $postDataHelper->getPostData($block->escapeUrl
+                                                            ($block->getAddToCartUrl($_item)),
+                                                                ['product' => $_item->getEntityId()])
+                                                        ?>
```

Comment:
Please separate logic from, view and don't use helpers inside of phtml

Reply:
Hi @Stepa4man 
thanks for your comment .. 
for this I have just adjusted the already written code .. 
but I will manage it from outside phtml